include/tools.h

```diff
@@ -214,6 +214,8 @@ bool fileExists(const std::string& path);
  */
 bool fileReadable(const std::string& path);
 
+bool makeDirectory(const std::string& path);
```

Comment:
Now I wonder if we need `makeDirectory()` in `libkiwix` at all. In `kiwix-desktop` we can use [`QDir::mkpath()`](https://doc.qt.io/qt-5/qdir.html#mkpath). So if we require that `sessionFileDir` refers to an existing directory we can drop `makeDirectory()` as well.

Reply:
> So if we require that sessionFileDir refers to an existing directory we can drop makeDirectory() as well.

What I am thinking of in regards to that is if we are ok on the first run.

From what I see the only call to `makeDirectory()` derives from `createDownloader()` which uses `getDataDirectory()`. So if we use `QDir::mkpath()` in that we should be ok, so I will go on with the removal of this also.